src/main/java/com/example/animalshelter/service/AnimalNotFoundException.java

```diff
@@ -0,0 +1,20 @@
+package com.example.animalshelter.service;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+/**
+ * Throw this exception when delete action failed.
+ * Like when animal id is not present id database.
+ */
+@ResponseStatus(HttpStatus.NOT_FOUND)
+public class AnimalNotFoundException extends Exception {
```

Comment:
Let's extend RuntimeException as it is not checked. There is no animal and you cannot do anything about it.

Reply:
Done.